cdc/kv/region_worker_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kv
+
+import (
+	"math/rand"
+	"sync"
+
+	"github.com/pingcap/check"
+	"github.com/pingcap/ticdc/pkg/util/testleak"
+)
+
+type regionWorkerSuite struct{}
+
+var _ = check.Suite(&regionWorkerSuite{})
+
+func (s *regionWorkerSuite) TestRegionStateManager(c *check.C) {
+	defer testleak.AfterTest(c)()
+	rsm := newRegionStateManager(4)
+
+	regionID := uint64(1000)
+	_, ok := rsm.getState(regionID)
+	c.Assert(ok, check.IsFalse)
+
+	rsm.setState(regionID, &regionFeedState{requestID: 2})
+	state, ok := rsm.getState(regionID)
+	c.Assert(ok, check.IsTrue)
+	c.Assert(state.requestID, check.Equals, uint64(2))
+}
+
+func (s *regionWorkerSuite) TestRegionStateManagerThreadSafe(c *check.C) {
+	defer testleak.AfterTest(c)()
+	rsm := newRegionStateManager(4)
+	regionCount := 100
+	regionIDs := make([]uint64, regionCount)
+	for i := 0; i < regionCount; i++ {
+		regionID := uint64(1000 + i)
+		regionIDs[i] = regionID
+		rsm.setState(regionID, &regionFeedState{requestID: uint64(i + 1), lastResolvedTs: uint64(1000)})
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
```

Comment:
could you please add more goroutine to make sure the thread-safety?

Reply:
addressed